src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/HardcodedSymmetricAlgorithmKeysSources.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    internal static class HardcodedSymmetricAlgorithmKeysSources
+    {
+        /// <summary>
+        /// <see cref="SourceInfo"/>s for hardcoded symmetric algorithm cryptographic keys tainted data sources.
+        /// </summary>
+        public static ImmutableHashSet<SourceInfo> SourceInfos { get; }
+
+        /// <summary>
+        /// Statically constructs.
+        /// </summary>
+        static HardcodedSymmetricAlgorithmKeysSources()
```

Comment:
I thought it was better to avoid the use of static ctor when we have an alternative? Here we could simply to something like:

```csharp
public static ImmutableHashSet<SourceInfo> SourceInfos { get; } = BuildSources();

private static ImmutableHashSet<SourceInfo> BuildSource()
{
	// Copy code from static ctor.
}
```

Reply:
What are the disadvantages of a static constructor here?

Reply:
@dotpaul Probably [CA1810](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1810), not sure.

Reply:
Ah, got it. I don't think the performance of a static initialization check is bad since these are accessed one or a few times per compilation, but it's easy enough to follow @Evangelink's suggestion for a best practice. I'll save other sources / sanitizers / sinks for another PR.